src/Templates/BlockTemplatesCompatibility.php

```diff
@@ -0,0 +1,290 @@
+			return $block_content;
+		}
+
+		/**
+		 * If the block is empty, we don't need to inject hooks.
+		 */
+		if ( empty( $block_content ) ) {
+			return $block_content;
+		}
+
+		$hooks = $hook_data[ $block_name ];
+
+		return sprintf(
+			'%1$s%2$s%3$s',
+			$this->get_hooks_buffer( $hooks, 'before' ),
+			$block_content,
+			$this->get_hooks_buffer( $hooks, 'after' )
+		);
+	}
+
+	/**
+	 * Hoding the supported hooks, their positions and the hooked functions.
+	 *
+	 * @return array Hook data.
+	 */
+	protected function get_hook_data() {
+		$hook_data = array(
+			'core/query'            => array(
+				'woocommerce_before_main_content' => array(
+					'position' => 'before',
+					'hooked'   => array(
+						'woocommerce_output_content_wrapper' => 10,
+						'woocommerce_breadcrumb' => 20,
+					),
+				),
+				'woocommerce_after_main_content'  => array(
+					'position' => 'after',
+					'hooked'   => array(
+						'woocommerce_output_content_wrapper_end' => 10,
+					),
+				),
+			),
+			'core/post-title'       => array(
+				'woocommerce_before_shop_loop_item_title' => array(
+					'position' => 'before',
+					'hooked'   => array(
+						'woocommerce_show_product_loop_sale_flash' => 10,
+						'woocommerce_template_loop_product_thumbnail' => 10,
+					),
+				),
+				'woocommerce_shop_loop_item_title'        => array(
+					'position' => 'after',
+					'hooked'   => array(
+						'woocommerce_template_loop_product_title' => 10,
+					),
+				),
+				'woocommerce_after_shop_loop_item_title'  => array(
+					'position' => 'before',
+					'hooked'   => array(
+						'woocommerce_template_loop_rating' => 5,
+						'woocommerce_template_loop_price'  => 10,
+					),
+				),
+			),
+			self::LOOP_ITEM_ID      => array(
+				'woocommerce_before_shop_loop_item' => array(
+					'position' => 'before',
+					'hooked'   => array(
+						'woocommerce_template_loop_product_link_open' => 10,
+					),
+				),
+				'woocommerce_after_shop_loop_item'  => array(
+					'position' => 'after',
+					'hooked'   => array(
+						'woocommerce_template_loop_product_link_close' => 5,
+						'woocommerce_template_loop_add_to_cart' => 10,
+					),
+				),
+			),
+			'core/post-template'    => array(
+				'woocommerce_before_shop_loop' => array(
+					'position' => 'before',
+					'hooked'   => array(
+						'woocommerce_output_all_notices' => 10,
+						'woocommerce_result_count'       => 20,
+						'woocommerce_catalog_ordering'   => 30,
+					),
+				),
+				'woocommerce_after_shop_loop'  => array(
+					'position' => 'after',
+					'hooked'   => array(
+						'woocommerce_pagination' => 10,
+					),
+				),
+			),
+			'core/query-no-results' => array(
+				'woocommerce_no_products_found' => array(
+					'position' => 'before',
+					'hooked'   => array(
+						'wc_no_products_found' => 10,
+					),
+				),
+			),
+		);
+
+		/**
+		 * Filter the hook data used to replace the default WooCommerce hooks.
+		 *
+		 * @param array $hook_data Hook data.
+		 */
```

Comment:
Given the complexity of the hook data array, it'd be good to expand on the doc block a bit more about its shape.